main.cpp

```diff
@@ -133,256 +175,335 @@ struct CarWash
         std::string color = "Green";
         bool readyToEat = false;
         int numberOfSeeds = 5;
-        // member functions
-        int produceSeeds(); // returns number of seeds produced
+
+        int produceSeeds();
         void fallFromTree();
-        void inspireTheoryOfGravity(); // did this actually happen with Newton?
+        void inspireTheoryOfGravity();
     };
-    // 3 things it can do:
-    // 1) grow
-    int grow(int sunlight, int water); // returns growth amount
-    // 2) produce fruit
+
+    int grow(int sunLight, int water);
     Fruit produceFruit();
-    // 3) photosynthesize
-    void photosynthesize();
+    int photosynthesize(int sunLight, int water, int unkonwnFactor);
 };
 
-struct Person
+int Tree::grow(int sunLight, int water)
+{
+    return sunLight * water;
+}
+// I couldn't quite figure out how to make this return a Fruit 
+// Tree::Fruit produceFruit(int grow)
+// {
+//      if (grow > 0)
+//      {
+//          return
+//             bool readyToEat = true;
+//      }
+//     else 
+//      {      
+//         return
+//             bool readyToEat = false;
+//     }
+// }
+
+int Tree::photosynthesize(int sunLight, int water, int unkonwnFactor)
+{
+    return (sunLight > 0) ? water * unkonwnFactor : 0;
+}
+
+struct PersonTwo
 {
-    // height
     float height = 5.4f;
-    // age
     int age = 29;
-    // number of siblings
     int numberOfSiblings = 3;
-    // job
     std::string job = "Driver";
-    // address
     std::string address = "1 S Anywhere St";
 
-    // Nested UDT
     struct Eye
     {
-        // member variables
         std::string color = "Blue";
         bool needsGlasses = true;
         bool hasCataracts = false;
         std::string typeOfColorBlindness = "Blue/ green";
         std::string vision = "20/50";
-        // member functions
-        void open();
-        void close();
+
+        bool open(bool awake);
+        bool close(bool awake);
         void see(std::string vision, std::string typeOfColorBlindness);
     };
-    // 3 things it can do:
-    // 1) run
-    void run(int speed);
-    // 2) jump
-    void jump(int height);
-    // 3) play piano
-    void playPiano(std::string piece);
+
+    bool run(int speed);
+    int jump(int crouch, int energy);
+    bool playPiano(bool memoryRecal, int nerves);
 };
 
+bool PersonTwo::run(int speed)
+{
+    return (speed > 5) ? true : false;
+}
+
+int PersonTwo::jump(int crouch, int energy)
+{
+    return crouch * energy;
+}
+
+bool PersonTwo::playPiano(bool memoryRecal, int nerves)
+{
+     return memoryRecal && nerves > 10 ? true : false; 
+}
+
+bool PersonTwo::Eye::open(bool awake)
+{
+    return awake ? true : false;
+}
+
+bool PersonTwo::Eye::close(bool awake)
+{
+    return awake == false ? true : false;
+}
+
+// void myPerson::Eye::see(std::string vision, std::string typeOfColorBlindness)
+// {
+        // I couldn't quite figure this one out.  Maybe I can get some pointers for my code review
+// }
```

Comment:
What do you want this function to do?  It's ok for the function to be empty at this time in the project.
- [ ] revise

Reply:
Sorry, I should have elaborated before.  I want the function to have the person see based on the relevant attributes of the eye struct